bpf/lib/proxy_hairpin.h

```diff
@@ -0,0 +1,89 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (C) 2021 Authors of Cilium */
+
+#ifndef __LIB_PROXY_HAIRPIN_H_
+#define __LIB_PROXY_HAIRPIN_H_
+
+#include "common.h"
+#include "utils.h"
+#include "ipv6.h"
+#include "ipv4.h"
+#include "eth.h"
+#include "dbg.h"
+#include "trace.h"
+#include "csum.h"
+#include "l4.h"
+
+/**
+ * ctx_redirect_to_proxy_hairpin redirects to the proxy by hairpining the
+ * packet out the incoming interface
+ */
+static __always_inline int
+ctx_redirect_to_proxy_hairpin(struct __ctx_buff *ctx, __be16 proxy_port)
```

Comment:
Let's add a suffix to the IPv4 version too?
```suggestion
ctx_redirect_to_proxy_hairpin_ipv4(struct __ctx_buff *ctx, __be16 proxy_port)
```

Reply:
Done :-)